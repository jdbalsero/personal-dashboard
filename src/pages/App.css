/* Variables globales */
:root {
  --main-blue-dashed-color: #7C93FE;
  --main-blue-color: #5873FE;
}

@font-face {
  font-family: 'borna';
  src: url('../assets/fonts/borna/borna-medium.otf') format('opentype');
}

@font-face {
  font-family: 'aeonik';
  src: url('../assets/fonts/aeoniktrial/AeonikTRIAL-Regular.otf') format('opentype');
}

@font-face {
  font-family: 'nib';
  src: url('../assets/fonts/nib/NibPro-Regular.woff2') format('opentype');
}

body, input, button {
  font-family: 'aeonik', sans-serif;
}

h1, h2, h3 {
  font-family: 'borna', sans-serif;
  margin: 10px;
}

body{
  overflow: hidden;
  margin: 0;
  height: 100vh;
}

#root {
  height: 100vh;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

.blur-class {
  filter: blur(5px);
}

.popup-sweet {
  border-radius: 20px;
  padding: 30px;
}

.title-sweet {
  color: #000;
  text-align: center;
  font-style: normal;
  font-size: 23px!important;
  font-weight: bold;
  line-height: normal;
}

.text-sweet {
  color: #000!important;
  text-align: center;
  font-family: 'aeonik';
  font-size: 30px;
  font-style: normal;
  font-weight: 300!important;
  line-height: 22px!important;
}

.deny-button-sweet:hover, .confirm-button-sweet:hover {
  background-color: var(--main-blue-color)!important;
  color: #FFF!important;
}

.deny-button-sweet, .confirm-button-sweet {
  background-color: #FFF!important;
  border: 3px solid var(--main-blue-color)!important;
  color: #000!important;
  border-radius: 8px!important;
  font-weight: bold!important;
}