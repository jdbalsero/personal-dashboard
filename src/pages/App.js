import { useState, useEffect } from 'react';
import './App.css';

import { Route, Switch, BrowserRouter } from "react-router-dom";
import { MainPage } from '../components/MainPage';

import { PageProvider } from '../context/GlobalContext';
// import { RegisterSteps } from './register/registerSteps';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import { PrivateRoute } from '../components/Shared/PrivateRoute';
// import { Login } from './login';
// import { ApproveTask } from './clientContent/approveTask';
// import { ForgotPassword } from './register/registerSteps/ForgotPassword';

function App() {
    return (
        <LocalizationProvider dateAdapter={AdapterDayjs}>
            <BrowserRouter>
                <PageProvider>
                    <Switch>
                        <Route exact path='/' component={MainPage} />
                        <PrivateRoute exact path='/:component' component={MainPage} />
                        {/* <Route exact path='/approveTask' component={ApproveTask} />
                        <Route exact path='/register' component={RegisterSteps} />
                        <Route exact path='/forgotPassword' component={ForgotPassword} />
                        <Elements stripe={stripePromise}>
                            <PrivateRoute exact path='/user/:component' component={MainPage} />
                        </Elements> */}
                    </Switch>
                </PageProvider>
            </BrowserRouter>
        </LocalizationProvider>
    );
}

export default App;
