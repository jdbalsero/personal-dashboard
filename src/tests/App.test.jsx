import App from '../pages/App';
import { render } from '@testing-library/react';


describe('Principal App Test', () => {

    test('Match Initial Snapshot', () => {
        const { container } = render(<App />);
        expect( container ).toMatchSnapshot();
    });
});