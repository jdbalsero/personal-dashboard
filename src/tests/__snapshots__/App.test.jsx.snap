// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Principal App Test Match Initial Snapshot 1`] = `
<div>
  <div
    class="main-page"
  >
    <div
      class="navbar"
    >
      <div
        class="left-navbar"
      >
        <a
          href="/Weather"
        >
          <div
            class="navbar-logo-container"
          >
            <img
              alt="Assisttu Logo"
              class="NavbarLogo"
              src="dashboard-logo.png"
            />
          </div>
        </a>
      </div>
    </div>
    <div
      class="content-wrapper"
    >
      <div
        class="sidebar "
      >
        <div
          class="upper-sidebar "
        >
          <ul
            class="ListSidebar MainListSidebar "
          >
            <li>
              <button
                class="ButtonSidebar  "
              >
                <div
                  class="ButtonIcon"
                >
                  <svg
                    baseProfile="tiny"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M14.5 3l-1 3-1-3c-.184-.553.114-1.149.666-1.333.553-.185 1.15.114 1.334.666.075.226.07.458 0 .667zM19.864 6.05l-2.829 1.415 1.415-2.829c.261-.521.894-.731 1.414-.472.521.261.731.894.472 1.415-.107.212-.274.372-.472.471zM21.5 12l-3-1 3-1c.553-.185 1.149.114 1.334.667.184.552-.115 1.148-.668 1.333-.225.075-.457.069-.666 0zM8.55 4.636l1.415 2.829-2.829-1.415c-.521-.261-.732-.894-.472-1.414.261-.521.895-.731 1.414-.472.213.107.373.274.472.472zM17.776 12.342c.139-.424.224-.871.224-1.342 0-2.481-2.019-4.5-4.5-4.5-1.34 0-2.537.594-3.357 1.528l-.143-.028c-1.776 0-3.369.78-4.469 2.011-.24-.08-.472-.086-.697-.011-.553.185-.852.781-.668 1.333.057.167.158.299.277.411-.283.697-.443 1.458-.443 2.256l.002.126c-1.725.445-3.002 2.013-3.002 3.874 0 2.206 1.795 4 4 4h11c2.757 0 5-2.243 5-5 0-2.129-1.344-3.939-3.224-4.658zm-4.276-3.842c1.379 0 2.5 1.121 2.5 2.5 0 .366-.096.706-.238 1.019-.354.021-.72.074-1.118.188-.521-1.353-1.604-2.415-2.967-2.905.456-.49 1.102-.802 1.823-.802zm2.5 11.5h-11c-1.104 0-2-.897-2-2s.896-2 1.908-2.006l1.422.016-.248-1.202c-.055-.263-.082-.536-.082-.808 0-2.206 1.795-4 4-4l.069-.014c1.904.055 3.495 1.406 3.847 3.27l.038.186c.123.436.517.706.946.712l.289-.023c.312-.09.569-.131.811-.131 1.654 0 3 1.346 3 3s-1.346 3-3 3z"
                      />
                    </g>
                  </svg>
                </div>
                <div
                  class="ButtonText"
                >
                  Weather
                </div>
              </button>
            </li>
            <li>
              <button
                class="ButtonSidebar  "
              >
                <div
                  class="ButtonIcon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="96"
                      rx="16"
                      ry="16"
                      width="96"
                      x="96"
                      y="112"
                    />
                    <path
                      d="M468 112h-52v304a32 32 0 0032 32 32 32 0 0032-32V124a12 12 0 00-12-12z"
                    />
                    <path
                      d="M431.15 477.75A64.11 64.11 0 01384 416V44a12 12 0 00-12-12H44a12 12 0 00-12 12v380a56 56 0 0056 56h342.85a1.14 1.14 0 00.3-2.25zM96 208v-96h96v96zm224 192H96v-32h224zm0-64H96v-32h224zm0-64H96v-32h224zm0-64h-96v-32h96zm0-64h-96v-32h96z"
                    />
                  </svg>
                </div>
                <div
                  class="ButtonText"
                >
                  News
                </div>
              </button>
            </li>
            <li>
              <button
                class="ButtonSidebar  "
              >
                <div
                  class="ButtonIcon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M139.61 35.5a12 12 0 0 0-17 0L58.93 98.81l-22.7-22.12a12 12 0 0 0-17 0L3.53 92.41a12 12 0 0 0 0 17l47.59 47.4a12.78 12.78 0 0 0 17.61 0l15.59-15.62L156.52 69a12.09 12.09 0 0 0 .09-17zm0 159.19a12 12 0 0 0-17 0l-63.68 63.72-22.7-22.1a12 12 0 0 0-17 0L3.53 252a12 12 0 0 0 0 17L51 316.5a12.77 12.77 0 0 0 17.6 0l15.7-15.69 72.2-72.22a12 12 0 0 0 .09-16.9zM64 368c-26.49 0-48.59 21.5-48.59 48S37.53 464 64 464a48 48 0 0 0 0-96zm432 16H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                    />
                  </svg>
                </div>
                <div
                  class="ButtonText"
                >
                  Tasks
                </div>
              </button>
            </li>
            <li>
              <button
                class="ButtonSidebar  "
              >
                <div
                  class="ButtonIcon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.445 11.688V6.354h-.633A12.6 12.6 0 0 0 4.5 7.16v.695c.375-.257.969-.62 1.258-.777h.012v4.61h.675zm1.188-1.305c.047.64.594 1.406 1.703 1.406 1.258 0 2-1.066 2-2.871 0-1.934-.781-2.668-1.953-2.668-.926 0-1.797.672-1.797 1.809 0 1.16.824 1.77 1.676 1.77.746 0 1.23-.376 1.383-.79h.027c-.004 1.316-.461 2.164-1.305 2.164-.664 0-1.008-.45-1.05-.82h-.684zm2.953-2.317c0 .696-.559 1.18-1.184 1.18-.601 0-1.144-.383-1.144-1.2 0-.823.582-1.21 1.168-1.21.633 0 1.16.398 1.16 1.23z"
                    />
                    <path
                      d="M3.5 0a.5.5 0 0 1 .5.5V1h8V.5a.5.5 0 0 1 1 0V1h1a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V3a2 2 0 0 1 2-2h1V.5a.5.5 0 0 1 .5-.5zM1 4v10a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V4H1z"
                    />
                  </svg>
                </div>
                <div
                  class="ButtonText"
                >
                  Calendar
                </div>
              </button>
            </li>
          </ul>
        </div>
        <div
          class="lower-sidebar"
        />
        <div
          class="collapse-button-container"
        >
          <button
            class="collapse-button"
          >
            <svg
              class="collapse-icon"
              color="#A4A4A4"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(164, 164, 164);"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 48C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48zm43.4 289.1c7.5 7.5 7.5 19.8 0 27.3-3.8 3.8-8.7 5.6-13.6 5.6s-9.9-1.9-13.7-5.7l-94-94.3c-6.9-7.6-6.7-19.3.6-26.6l95.4-95.7c7.5-7.5 19.7-7.6 27.3 0 7.5 7.5 7.6 19.7 0 27.3l-81.9 81 79.9 81.1z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
